test/syntax.jl

```diff
@@ -3357,3 +3351,25 @@ demo44723()::Any = Base.Experimental.@opaque () -> true ? 1 : 2
     @test y == Core.svec(2, 3)
     @test z == 4
 end
+
+@testset "issue 25072" begin
+    @test '\xc0\x80' == reinterpret(Char, 0xc0800000)
+    @test '\x80' == reinterpret(Char, 0x80000000)
+    @test '\xff' == reinterpret(Char, 0xff000000)
+    @test_parseerror "'\\xff\\xff\\xff\\xff'" "character literal contains multiple characters" # == reinterpret(Char, 0xffffffff)
+    @test '\uffff' == Char(0xffff)
+    @test '\U00002014' == Char(0x2014)
+    @test '\100' == reinterpret(Char, UInt32(0o100) << 24)
```

Comment:
This is just `@`, right?

Reply:
Yeah, perhaps this makes it more obvious:

```suggestion
    @test '\100' == reinterpret(Char, UInt32(0o100) << 24) == '@'
```